File: jsparser/jsparser.go
package jsparser

import (
	"os"
	"os/exec"
	"regexp"
)

func Parse(jsURL string) {
	cmd := exec.Command("curl", "-s", jsURL)
	out, err := cmd.Output()
	if err != nil {
		return
	}

	re := regexp.MustCompile(`["'](/api/[^"']+)["']`)
	matches := re.FindAllStringSubmatch(string(out), -1)

	f, _ := os.OpenFile("output/js_endpoints.txt", os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
	defer f.Close()

	for _, m := range matches {
		f.WriteString(m[1] + "\n")
	}
}

File: core/js_parser.py
import subprocess
import re

def parse_js(js_url):
    try:
        js = subprocess.check_output(
            ["curl", "-s", js_url],
            universal_newlines=True
        )

        endpoints = set(re.findall(r'["\'](/api/[^"\']+)["\']', js))

        with open("output/js_endpoints.txt", "a") as f:
            for ep in endpoints:
                f.write(ep + "\n")

    except:
        pass
